azurerm/helpers/azure/resourceid.go

```diff
@@ -0,0 +1,154 @@
+package azure
+
+import (
+	"fmt"
+	"net/url"
+	"sort"
+	"strings"
+)
+
+// ResourceID represents a parsed long-form Azure Resource Manager ID
+// with the Subscription ID, Resource Group and the Provider as top-
+// level fields, and other key-value pairs available via a map in the
+// Path field.
+type ResourceID struct {
+	SubscriptionID string
+	ResourceGroup  string
+	Provider       string
+	Path           map[string]string
+}
+
+// parseAzureResourceID converts a long-form Azure Resource Manager ID
+// into a ResourceID. We make assumptions about the structure of URLs,
+// which is obviously not good, but the best thing available given the
+// SDK.
+func ParseAzureResourceID(id string) (*ResourceID, error) {
```

Comment:
are we also going to call this method from the existing method to give us a migration path whilst refactoring?

Reply:
Thats a good idea, now threading down to new method